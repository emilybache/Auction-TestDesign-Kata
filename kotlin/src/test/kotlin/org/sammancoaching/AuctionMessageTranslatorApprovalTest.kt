package org.sammancoaching

import org.approvaltests.Approvals
import kotlin.test.Test

class AuctionMessageTranslatorApprovalTest {
    @Test
    fun notifiesAuctionClosedWhenCloseMessageReceived() {
        val spy = StringBuilder()
        val message = "SOLVersion: 1.1; Event: CLOSE;"
        spy.append("Processing event: $message\n")
        val listener = AuctionEventListenerApprovalSpy(spy)
        val translator = AuctionMessageTranslator(listener)

        translator.processMessage(message)

        Approvals.verify(spy)
    }

    @Test
    fun notifiesBidDetailsWhenPriceMessageReceived() {
        val spy = StringBuilder()
        val message = "SOLVersion: 1.1; Event: PRICE; CurrentPrice: 192; Increment: 7; Bidder: Someone else;"
        spy.append("Processing event: $message\n")
        val listener = AuctionEventListenerApprovalSpy(spy)
        val translator = AuctionMessageTranslator(listener)

        translator.processMessage(message)

        Approvals.verify(spy)
    }
}