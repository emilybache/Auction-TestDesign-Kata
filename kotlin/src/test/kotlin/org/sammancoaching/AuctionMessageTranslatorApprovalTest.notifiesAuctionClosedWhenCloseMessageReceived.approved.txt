Processing event: SOLVersion: 1.1; Event: CLOSE;
auctionClosed